subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/AttributeContainerResolutionIntegrationTest.groovy

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Issue
+
+class AttributeContainerResolutionIntegrationTest extends AbstractIntegrationSpec {
+    @Issue("https://github.com/gradle/gradle/issues/26298")
+    def "lazy attributes provided to a Configuration do not fail resolution when they have side effects"() {
+        settingsFile << """
+            include "sub"
+        """
+        file("sub/build.gradle") << """
+            configurations {
+                consumable("conf") {
+                    attributes {
+                        attribute(Attribute.of("attr", Named), objects.named(Named, "value"))
+                    }
+                }
+            }
+        """
+        buildFile << """
+            configurations {
+                conf {
+                    attributes {
+                        def otherAttribute = Attribute.of("zzz", Named)
+                        attributeProvider(Attribute.of("aaa", Named), provider {
+                            assert getAttribute(otherAttribute).name == "other"
+                            objects.named(Named, "value")
+                        })
+                        attributeProvider(otherAttribute, provider {
+                            objects.named(Named, "other")
+                        })
+                    }
+                }
+            }
+
+            dependencies {
+                conf(project(":sub"))
```

Comment:
Does this test need to have a second local project to reproduce the issue? Or can this also be reproduced by depending on an external dependency?

Reply:
I didn't think about trying this without any dependencies, so I can try that.  I just wanted to avoid the short-circuiting of resolution. 

An external dependency would require a repository (either maven central or something for the test only) and I didn't want to faff with that.

Reply:
Yeah the short circuit would get in the way.

The maven repository test fixture is surprisingly easy to use. It takes one line to publish a fake module and then another line to register that test repo with the project. It is also available on `AbstractIntegrationSpec` so you don't need to extend any new interfaces or declare any Rule fields